### PR TITLE
Partition: handle loop devices and multiple mounts per device

### PR DESCRIPTION
**Pull request by clebergnu**
This includes the original change from @harish-24 , along with a fix in the handling of loop devices.

### CODE REVIEW

File: avocado/utils/partition.py [hunk 1]
```diff
@@ -298,18 +300,27 @@ def unmount(self, force=True):
         """
         with MtabLock():
             mountpoint = self.get_mountpoint()
```
Comment (harish-24) [inline avocado/utils/partition.py:302]: One more catch here: if any test uses `mount()` method with a different mountpoint (optional arg `mountpoint`) then we fail to get that mountpoint here and hence there is a chance unmount will fail. If that is the case we should either restrict taking an alternate mountpoint in mount method or have an optional `mountpoint` arg specified here in `unmount()` method.

Is that fair or am I missing something?

**Comment by clebergnu** [comment]
The mountpoint used by default is the one set as the instance attribute, so
IIUC it doesn't need to be an extra parameter and the right one should be
found and returned.

**Comment by harish-24** [comment]
> The mountpoint used by default is the one set as the instance attribute, so IIUC it doesn't need to be an extra parameter and the right one should be found and returned.
> […](#)

okay so we can assume that the testcase is responsible for providing the default mountpoint when init of the class object and not during calling the mount() function?

**Comment by clebergnu** [comment]
> > The mountpoint used by default is the one set as the instance attribute, so IIUC it doesn't need to be an extra parameter and the right one should be found and returned.
> > […](#)
> 
> okay so we can assume that the testcase is responsible for providing the default mountpoint when init of the class object and not during calling the mount() function?

OK, now I see your point.  Yes, tests currently rely on the mountpoint given when instantiating the Partition object, so they will use the "default" (as in the instance attribute's) mountpoint.  I now see that the API for mount allows for a different mountpoint, while the API for unmount doesn't.

That's inconsistent indeed.  But I believe it's a different problem.

**Review by willianrampazzo — APPROVED** [review]
Considering the concern of @harish-24 will be handled in a different PR, LGTM.

**Comment by clebergnu** [comment]
> Considering the concern of @harish-24 will be handled in a different PR, LGTM.

I've created an issue to track it: #4463

**Pull request merged**